### PR TITLE
Add version to the binary during build

### PR DESCRIPTION
**Pull request by umbynos**
Replace the string in the patch overriding the version, This way when running `imgtool version` the correct version is reported.

### CODE REVIEW

**Comment by umbynos** [comment]
Example of the problem can be seen here https://github.com/arduino/imgtool-packing/pull/11#issuecomment-1125061411

**Review by per1234 — COMMENTED** [review]
What do you think about using an obvious placeholder in the patch to make the system more clear?

Like changing the current patch:

https://github.com/arduino/imgtool-packing/blob/5619cae603dfcef2870e66f510135b032979a828/patches/0008-Imgtool-Append-arduino-to-version-string.patch#L18-L19

To something like this:

```patch
-imgtool_version = "1.8.0"
+imgtool_version = "ARDUINO_VERSION_PLACEHOLDER"
```

**Review by per1234 — APPROVED** [review]

**Review by per1234 — CHANGES_REQUESTED** [review]
I forgot I was going to make one follow-up suggestion if the placeholder idea was accepted:

Update the patch message to reflect the new nature of the patch:

https://github.com/arduino/imgtool-packing/blob/2fd18071dbab7179fc95f2d23a646a8bf91a55c2/patches/0008-Imgtool-Append-arduino-to-version-string.patch#L4

**Comment by umbynos** [comment]
Test release here https://github.com/arduino/imgtool-packing/releases/tag/1.8.0-rc7

**Comment by umbynos** [comment]
```
❰umberto❙~/Downloads/imgtool_1.8.0-rc7_Linux_64bit❱✔≻ ./imgtool version                                                              11:11:49
1.8.0-rc7
```

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -55,10 +55,14 @@ jobs:
         uses: actions/checkout@v3
         with:
           path: ${{ env.IMGTOOL_PACKING_PATH }}
+        
+      - name: Set the version
+        working-directory: ${{ env.IMGTOOL_PACKING_PATH }}/patches/
+        run: perl -pi.bak -e "s/ARDUINO_VERSION_PLACEHOLDER/${GITHUB_REF/refs\/tags\//}/g" 0008-Imgtool-Append-arduino-to-version-string.patch
```
Comment (per1234) [inline .github/workflows/release.yml:61]:
```suggestion
        run: perl -pi -e "s/ARDUINO_VERSION_PLACEHOLDER/${GITHUB_REF/refs\/tags\//}/g" 0008-Imgtool-Append-arduino-to-version-string.patch
```
There is no need to make a backup file that won't be used.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -134,9 +138,13 @@ jobs:
         with:
           path: ${{ env.IMGTOOL_PACKING_PATH }}
 
+      - name: Set the version
+        working-directory: ${{ env.IMGTOOL_PACKING_PATH }}/patches/
+        run: sed -i "s/ARDUINO_VERSION_PLACEHOLDER/${GITHUB_REF/refs\/tags\//}/" 0008-Imgtool-Append-arduino-to-version-string.patch
```
Comment (per1234) [inline .github/workflows/release.yml:143]:
```suggestion
        run: perl -pi -e "s/ARDUINO_VERSION_PLACEHOLDER/${GITHUB_REF/refs\/tags\//}/g" 0008-Imgtool-Append-arduino-to-version-string.patch
```
Using a consistent placeholder replacement command throughout the workflow will make it easier to understand and maintain.

`sed` and `perl` are confusing enough on their own for dummies like me without having to figure both of them out at the same time!

**Review by per1234 — CHANGES_REQUESTED** [review]

**Review by per1234 — APPROVED** [review]

**Pull request merged**